### ISSUE TITLE
Proposal: Rename ServiceCatalogBroker to ServiceBroker

### ISSUE THREAD

**Issue by pmorie** [kubernetes-incubator/service-catalog#1060]
Per this comment: https://github.com/kubernetes-incubator/service-catalog/pull/1054#issuecomment-317460675, I think we can reasonably consume this name, and it's far less confusing (and less to type) than `ServiceCatalogBroker`.

Any thoughts?

**Comment by duglin** [comment]
while ServiceBroker by itself is ok, there was a desire to have all of our resources have the same prefix and “Service” wasn’t unique enough.  e.g. it might be odd to have ServiceCatalogInstance but ServiceBroker. Keep in mind, most people will never deal with ServiceBrokers. only admins. If anything “Instance”,  “ServiceClass” and “Plan” are the ones we should worry about w.r.t. UX

**Comment by pmorie** [comment]
@duglin I don't understand what benefit having the same prefix has, can you enlighten me?

In general, I think users will prefer typing shorter names -- I'm going to keep the conversations straight by opening issues for other resources.

**Comment by MHBauer** [comment]
Until there is a final definitive decision on the names, I favor a consistent long prefix for all resources.

**Comment by pmorie** [comment]
> Until there is a final definitive decision on the names, I favor a consistent long prefix for all resources.

That's totally cool.  I would prefer not to do any further renames in code until we have final agreement on all of the names.

**Comment by duglin** [comment]
please don't open separate issues for each resource - let's just discuss the entire thing here in this one. Otherwise we'll get mixed/crossed threads.

**Comment by pmorie** [comment]
> please don't open separate issues for each resource - let's just discuss the entire thing here in this one. Otherwise we'll get mixed/crossed threads.

The only way to avoid mixed/crossed threads is to have a distinct issue for each object, IMO.

**Comment by duglin** [comment]
I don't think it make sense to discuss each resource name in isolation - they're not independent to me.  Which relates to your other question... "what's the benefit of the same perfix?"... its a better UX because it helps tie all of the concepts together.  E.g. why call something "ServiceCatalogInstance" but its parent is called "ServiceBroker".  Why did the word "Catalog" vanish?  I can't explain that to someone.  "Because it sounded better" isn't much of a reason since "ServiceInstance" sounds better too. So does "Binding".  Oh wait...  :-)

**Comment by pmorie** [comment]
@duglin how about we discuss {broker, serviceclass, serviceplan} here, since those have 'has a' relationship at a couple levels.

**Comment by pmorie** [comment]
I think based on the comment I cited in the issue description, we could legitimately claim:

- `ServiceBroker`
- `ServiceClass`
- `ServicePlan`
- `ServiceInstance`

...and we had already agreed on `ServiceInstanceCredential`

That would be an internally-consistent set of names without collisions elsewhere in the k8s ecosystem, and is 7 characters less for users to type.

**Comment by duglin** [comment]
I'd like to back up a sec... what changed in the 5 minutes after the other PR was merged to cause this to be an issue?  I'm not objecting to changing names again it just seems kind of odd timing since we didn't even have things merged long enough to find any UX or technical issues.  Seems more like something that should have been brought up 5 minutes earlier.

**Comment by pmorie** [comment]
@duglin nothing changed, as far as I know.  I thought we left the design meeting where this was discussed as 'this is our opener', and expected to further refine things.  I was surprised that you opened the PR to do the rename, but it moves us forward by disambiguating the name of binding.  I'm happy to revert the PR if you want.

**Comment by MHBauer** [comment]
I left the design meeting as "let's change it to make it long and ugly" pending a decision from upstairs on what the final location is along with a process for deciding such.

**Comment by pmorie** [comment]
Okay, I think we had a miscommunication.  I'm going to revert #1054 and we can discuss in tomorrow's design meeting.

**Comment by duglin** [comment]
my take away from the design session was that yes it might change due to some future reviews from folks like api-machinery, but I didn't think that **we** would change our minds **5 minutes** after the PR was merged. Kind of defeats the purpose of getting an agreement during our design session.

**Comment by pmorie** [comment]
I merged the #1054 PR because it disambiguated the name of the Binding resource.  In retrospect, I should probably have labeled that PR do-not-merge instead of merging it.  Apologies for the confusion - I will be happy to help making the PR that represents our eventual consensus.

I viewed the `ServiceCatalog` prefix as a 'strawman to start discussing with k8s owners' (your words, if I remember correctly).  In my mind, everyone fully expected to re-evaluate the names immediately upon getting more information.

After you opened your pull request, we got more information from a k8s owner about being able to claim names that are far more user-friendly (and more in line with precedent for how resources in every other API group are named).  I think the important bit here is not agreement within the group we have today, but what is best for users and the ecosystem in general.  Since we have more information now, that alone is reason to rediscuss.

Even though our API is alpha, we should take care not to break users in multiple releases.  I think it is best that we contain breaking API changes, if possible, to a single release.  Based on our current discussions, I think this would contain:

- Rename of API resources
- Splitting services/plans into separate resources
